### PR TITLE
feat: add ValueUnit struct

### PR DESCRIPTION
**Pull request by intergalacticspacehighway**
## Summary:
Adds a new struct ValueUnit for https://github.com/facebook/react-native/pull/38559. This is a simple version of `YGValue` struct to hold values that have percentage and point as units.

## Changelog:
[INTERNAL] [ADDED] - ValueUnit struct

## Test Plan:
Pod install and build the rn-tester ios app

### CODE REVIEW

File: packages/react-native/ReactCommon/react/renderer/graphics/ValueUnit.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+namespace facebook::react {
+
+enum class UnitType {
+  UNDEFINED,
+  POINT,
+  PERCENT
```
Comment (NickGerleman) [inline packages/react-native/ReactCommon/react/renderer/graphics/ValueUnit.h:8]: nit: other Fabric enums are pascal case
```suggestion
  Undefined,
  Point,
  Percent,
```

File: packages/react-native/ReactCommon/react/renderer/graphics/ValueUnit.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+};
+
+struct ValueUnit {
+  float value;
+  UnitType unit;
+  
+  ValueUnit() : value(0.0f), unit(UnitType::UNDEFINED) {}
```
Comment (NickGerleman) [inline packages/react-native/ReactCommon/react/renderer/graphics/ValueUnit.h:15]: nit: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#c45-dont-define-a-default-constructor-that-only-initializes-data-members-use-in-class-member-initializers-instead

```suggestion
  float value{0.0f};
  UnitType unit{UnitType::Undefined};
```

**Comment by javache** [comment]
I think this PR should be "stacked" on the other transform-origin changes, so it can be introduced with an immediate use-case.

**Comment by intergalacticspacehighway** [comment]
@javache merged this branch here - https://github.com/facebook/react-native/pull/38559/commits/0f27a0c690e30de06ecc8a4473ca52feea8beb89 and usage - https://github.com/facebook/react-native/pull/38559/commits/a30ed3d9ca25be6a56f04508240e6ef052b204db. Feel free to close this PR!

**Pull request closed without merging**